src/views/Accounts/HardwareWallet/HardwareWallet.vue

```diff
@@ -189,42 +189,71 @@ export default {
       this.currentStep = 'token-management'
       this.selectedWalletType = walletType
     },
+    async createAccounts({ assetKeys, accounts, chain, name, accountType }) {
+      const assets = assetKeys.filter((asset) => {
+        return cryptoassets[asset]?.chain === chain
+      })
+
+      for (const key in accounts) {
+        const item = accounts[key]
+        const { publicKey, chainCode, derivationPath } = item
+        const index = item.index + 1
+
+        const account = {
+          name: `Ledger ${name} ${index}`,
+          alias: '',
+          chain,
+          addresses: [item.account],
+          assets,
+          index: item.index,
+          type: accountType,
+          enabled: true,
+          derivationPath,
+          color: getNextAccountColor(chain, item.index),
+          publicKey,
+          chainCode
+        }
+
+        await this.createAccount({
+          network: this.activeNetwork,
+          walletId: this.activeWalletId,
+          account
+        })
+      }
```

Comment:
@bradleySuira Well done man, Just a small change that we can make here to resole these promises more efficiently;

```
await Promise.all(accounts.map((key) => {
    const item = accounts[key]
        const { publicKey, chainCode, derivationPath } = item
        const index = item.index + 1
        const account = {
          name: `Ledger ${name} ${index}`,
          alias: '',
          chain,
          addresses: [item.account],
          assets,
          index: item.index,
          type: accountType,
          enabled: true,
          derivationPath,
          color: getNextAccountColor(chain, item.index),
          publicKey,
          chainCode
        }
        this.createAccount({
          network: this.activeNetwork,
          walletId: this.activeWalletId,
          account
        })
  }))
```

Reply:
we will iterate over the evm chains, taking your advice about the list or promises without any extra variable